operation/fhir-operation-validate/src/main/java/org/linuxforhealth/fhir/operation/validate/ValidateOperation.java

```diff
@@ -131,20 +120,88 @@ private OperationOutcome buildResourceValidOperationOutcome(List<Issue> issues)
     
     
     /**
-     * Check if a resource validation mode code is valid.
+     * This method does the following validations.
+     * 1. Validate if a resource validation mode code is valid.
+     * 2. Validate an interaction for a specified resource type.
+     * 3. Validate if persistence layer implementation supports update/create mode if mode = create.
+     * 4. Validate if persistence layer implementation supports update/create mode if mode = update but the resource doesnot exist yet in the DB.
+     * 5. Validate if modes update and delete are only be used when the operation is invoked at the resource instance level
      *
      * @param modeParameter resource validation mode code to be validated
-     * @throws FHIROperationException when the resource validation mode code is invalid
+     * @param resourceHelper Resource operation provider for loading related Library resources
+     * @param resourceType a valid resource type string
+     * @param operationContext the FHIROperationContext associated with the request
+     * @throws Exception 
+     */
+    private void validateModeParameter(ModeType modetype, FHIRResourceHelpers resourceHelper, String resourceType, FHIROperationContext operationContext, String logicalId) throws Exception {
+        if(modetype == null) {
+            return;
+        }
+        // Validate an interaction for a specified resource type.
+        if ((modetype == ModeType.CREATE  
+                || modetype == ModeType.UPDATE 
+                || modetype == ModeType.DELETE)
+                && resourceType != null) {
+            FHIRResourceHelpers.Interaction interaction = FHIRResourceHelpers.Interaction.from(modetype.value());
+            resourceHelper.validateInteraction(interaction, resourceType);
+        }
+        // Validate if persistence layer implementation supports update/create mode if mode = create.
+        if (operationContext != null && operationContext.getType() == FHIROperationContext.Type.INSTANCE 
+                && modetype == ModeType.CREATE) {
+            validateUpdateCreateEnabled(modetype, resourceType, operationContext);
```

Comment:
Hmm, the updateCreateEnabled flag is only relevant for UPDATE interactions. It indicates whether or not we can create a resource which doesn't yet exist.

Reply:
it sort of depends what mode=create means when invoked at the instance level.
if we take that to mean "can we do a create-on-update"? then i think the above is fine

Reply:
if we take it to mean something else, then mode=create should not even be a valid option when invoked at the instance level

Reply:
Right, CREATE at the instance level is a bit odd. I'm OK leaving it as is.